src/repr/src/row.rs

```diff
@@ -2826,11 +2832,11 @@ impl Default for RowArena {
 ///
 /// [`SharedRow::get`] panics when trying to obtain multiple references to the shared row.
 #[derive(Debug)]
-pub struct SharedRow(Rc<RefCell<Row>>);
+pub struct SharedRow(Row);
 
 impl SharedRow {
     thread_local! {
-        static SHARED_ROW: Rc<RefCell<Row>> = Rc::new(RefCell::new(Row::default()));
+        static SHARED_ROW: Cell<Option<Row>> = const { Cell::new(Some(Row::empty())) }
```

Comment:
might be worth leaving a note here, on lines 2850 and 2885 that the `Option` here is used a perf improvement to prevent `std::mem::take` from creating a new row when when it calls default in the underlying call to `std::mem::replace`. I'm not sure how obvious that would be here to people who are unfamiliar with take

Reply:
The `Option` here is not meant to be a performance improvement, and I would be surprised if the difference between `Cell<Option<Row>>` and `Cell<Row>` showed up in a benchmark. This is because calling calling `Row::default()` is very cheap, since it doesn't allocate.

The reason we need the `Option` here is because we want to track one bit of information, namely whether the shared row is currently borrowed. This allows us to catch any double borrows and panic whereas an `std::mem::take` would happily give back some `Row` every time.

Added a comment to clarify the structure

Reply:
gotcha, that makes sense!